pkg/sqlite/graphloader.go

```diff
@@ -0,0 +1,127 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+type SQLGraphLoader struct {
+	Querier     *SQLQuerier
```

Comment:
Can this accept a registry.Query instead of a *SQLQuerier and still do its job? What's tying it to the SQLite implementation? Either way, I'd like to see the NewX funcs accept the querier directly, instead of coupling the graph loader to the way queriers are created.

Reply:
The reason is that this entire loader implentation is "get the aggregation and abstraction that the sql layer provides and convert it into a graph". Literally all of the functions here are directly an abstraction around that. If we made it generic, then the input from other functions would have to be in the channelentry form, but that abstraction is exactly the one we are attempting to hide through this interface.

Reply:
Another example loader would be "okay now take some bundle images as input and load them into this struct directly" which just has a different set of concepts as input

Reply:
circling back around to this, because I misunderstood and talked to ben offline:

updated the struct to use the query interface for mocking purposes

Reply:
Looks good, thanks!